grove/measurements/estimation.py

```diff
@@ -147,7 +147,11 @@ def estimate_pauli_sum(pauli_terms, basis_transform_dict, program,
     pauli_for_rotations = PauliTerm.from_list(
         [(value, key) for key, value in basis_transform_dict.items()])
 
-    post_rotations = get_rotation_program(pauli_for_rotations)
+    program += get_rotation_program(pauli_for_rotations)
+
+    qubits = sorted(list(basis_transform_dict.keys()))
```

Comment:
This doesn't actually need to be sorted, right? 

Is this just so that the produced program is the same across different calls to this function?

Reply:
sorting was relevant for run_and_measure, see this comment:
https://github.com/rigetticomputing/grove/pull/161/files#diff-6bb0f8445c8a623a2a94eede5f3e094dL162

if it doesn't matter for run (i haven't tested this), i'm happy to remove it

Reply:
I see. I still think it's the case that the keys aren't promised to be in an order for python <3.7, and so we aren't guaranteed the produced program will be the same on different calls if we don't sort. Let's just leave it for now.